### PR TITLE
Added requestedTheme option into terminal settings

### PR DESCRIPTION
**Pull request by Gabboxl**
Added the possibility to change the theme (dark/light/default) from the terminal settings

### CODE REVIEW

**Comment by mdtauk** [comment]
The most common scenarios should be there in the GUI, with a button in the settings screen to open the commented JSON file with advanced settings

File: src/cascadia/TerminalApp/GlobalAppSettings.h [hunk 1]
```diff
@@ -42,6 +42,8 @@ class TerminalApp::GlobalAppSettings final
     bool GetShowTitleInTitlebar() const noexcept;
     void SetShowTitleInTitlebar(const bool showTitleInTitlebar) noexcept;
 
+	void SetRequestedTheme(const winrt::Windows::UI::Xaml::ElementTheme requestedTheme) noexcept;
```
Comment (gpotter2) [inline src/cascadia/TerminalApp/GlobalAppSettings.h]: You used tabs, whereas the convention is 4 spaces :)

Comment (Gabboxl) [reply src/cascadia/TerminalApp/GlobalAppSettings.h]: ops!

**Comment by DHowett-MSFT** [comment]
This change just exposes that setting in the default `profiles.json`, right? Without this change, if you _add_ `"requestedTheme": ...` it _will still work_, you just won't know the setting exists.

File: src/cascadia/TerminalApp/GlobalAppSettings.cpp [hunk 2]
```diff
@@ -96,6 +96,11 @@ ElementTheme GlobalAppSettings::GetRequestedTheme() const noexcept
     return _requestedTheme;
 }
 
+void GlobalAppSettings::SetRequestedTheme(const ElementTheme requestedTheme) noexcept
+{
+    _requestedTheme = requestedTheme;
+}
+
```
[the remark this answers is not in the document]
Comment (Gabboxl) [reply src/cascadia/TerminalApp/GlobalAppSettings.cpp]: done.

**Comment by Gabboxl** [comment]
@DHowett-MSFT yes, this change just exposes the setting to change the theme in the json file.

File: src/cascadia/TerminalApp/CascadiaSettings.cpp [hunk 3]
```diff
@@ -154,6 +154,8 @@ void CascadiaSettings::_CreateDefaultProfiles()
     _profiles.emplace_back(cmdProfile);
 
     _globals.SetDefaultProfile(powershellProfile.GetGuid());
+
+    _globals.SetRequestedTheme(winrt::Windows::UI::Xaml::ElementTheme::Default);
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/CascadiaSettings.cpp]: `CreateDefaultProfiles` isn't a great place for this .. is there something more generic or less default-profile-related to put it?

**Comment by DHowett-MSFT** [comment]
That makes sense. Thanks!

**Comment by Gabboxl** [comment]
I found what was blocking the serialization of the requestedTheme value in the json file. Now it works.

File: src/cascadia/TerminalApp/GlobalAppSettings.cpp [hunk 4]
```diff
@@ -147,11 +151,8 @@ JsonObject GlobalAppSettings::ToJson() const
 
     jsonObject.Insert(SHOW_TABS_IN_TITLEBAR_KEY,
                       JsonValue::CreateBooleanValue(_showTabsInTitlebar));
-    if (_requestedTheme != ElementTheme::Default)
-    {
         jsonObject.Insert(REQUESTED_THEME_KEY,
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/GlobalAppSettings.cpp]: would you mind fixing the indentation here, since you removed the enclosing scope?

Comment (Gabboxl) [reply src/cascadia/TerminalApp/GlobalAppSettings.cpp]: oh yes, right.

File: src/cascadia/TerminalApp/GlobalAppSettings.cpp [hunk 5]
```diff
@@ -147,11 +151,8 @@ JsonObject GlobalAppSettings::ToJson() const
 
     jsonObject.Insert(SHOW_TABS_IN_TITLEBAR_KEY,
                       JsonValue::CreateBooleanValue(_showTabsInTitlebar));
-    if (_requestedTheme != ElementTheme::Default)
-    {
-        jsonObject.Insert(REQUESTED_THEME_KEY,
+    jsonObject.Insert(REQUESTED_THEME_KEY,
                           JsonValue::CreateStringValue(_SerializeTheme(_requestedTheme)));
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/GlobalAppSettings.cpp]:
```suggestion
                      JsonValue::CreateStringValue(_SerializeTheme(_requestedTheme)));
```

:frown:

**Pull request merged**